Windows/MainWindow.cpp

```diff
@@ -365,63 +360,58 @@ namespace MainWindow
 	}
 
+	// Note that this also updates the config! Not very clean.
 	RECT DetermineWindowRectangle() {
-		RECT rc;
-
 		const int screenWidth = GetSystemMetrics(SM_CXVIRTUALSCREEN);
 		const int screenHeight = GetSystemMetrics(SM_CYVIRTUALSCREEN);
 		const int screenX = GetSystemMetrics(SM_XVIRTUALSCREEN);
 		const int screenY = GetSystemMetrics(SM_YVIRTUALSCREEN);
 
-		if (!g_Config.bFullScreen) {
+		bool resetPositionX = true;
+		bool resetPositionY = true;
+
+		if (g_Config.iWindowWidth > 0 && g_Config.iWindowHeight > 0 && !g_Config.bFullScreen) {
 			bool visibleHorizontally = ((g_Config.iWindowX + g_Config.iWindowWidth) >= screenX) &&
 				((g_Config.iWindowX + g_Config.iWindowWidth) < (screenWidth + g_Config.iWindowWidth));
 
 			bool visibleVertically = ((g_Config.iWindowY + g_Config.iWindowHeight) >= screenY) &&
 				((g_Config.iWindowY + g_Config.iWindowHeight) < (screenHeight + g_Config.iWindowHeight));
 
-			if (!visibleHorizontally)
-				g_Config.iWindowX = -1;
-
+			if (visibleHorizontally)
+				resetPositionX = false;
 			if (!visibleVertically)
-				g_Config.iWindowY = -1;
+				resetPositionY = false;
 		}
 
-		rc.left = g_Config.iWindowX;
-		rc.top = g_Config.iWindowY;
+		int windowWidth = g_Config.iWindowWidth;
+		int windowHeight = g_Config.iWindowHeight;
 
 		// First, get the w/h right.
-		if (g_Config.iWindowWidth <= 0 || g_Config.iWindowHeight <= 0) {
-			RECT rcInner = rc, rcOuter;
+		if (windowWidth <= 0 || windowHeight <= 0) {
 			bool portrait = g_Config.IsPortrait();
 
 			// We want to adjust for DPI but still get an integer pixel scaling ratio.
 			double dpi_scale = 96.0 / System_GetPropertyInt(SYSPROP_DISPLAY_DPI);
 			int scale = (int)ceil(2.0 / dpi_scale);
 
-			GetWindowRectAtResolution(scale * (portrait ? 272 : 480), scale * (portrait ? 480 : 272), rcInner, rcOuter);
-			rc.right = rc.left + (rcOuter.right - rcOuter.left);
-			rc.bottom = rc.top + (rcOuter.bottom - rcOuter.top);
-			g_Config.iWindowWidth = rc.right - rc.left;
-			g_Config.iWindowHeight = rc.bottom - rc.top;
-		} else {
-			rc.right = rc.left + g_Config.iWindowWidth;
-			rc.bottom = rc.top + g_Config.iWindowHeight;
+			GetWindowSizeAtResolution(scale * (portrait ? 272 : 480), scale * (portrait ? 480 : 272), &windowWidth, &windowHeight);
+			g_Config.iWindowWidth = windowWidth;
+			g_Config.iWindowHeight = windowHeight;
 		}
 
-		// Then center if necessary.
-		if (g_Config.iWindowX == -1 && g_Config.iWindowY == -1) {
-			// Center the window.
-			const int primaryScreenWidth = GetSystemMetrics(SM_CXSCREEN);
-			const int primaryScreenHeight = GetSystemMetrics(SM_CYSCREEN);
-			g_Config.iWindowX = (primaryScreenWidth - g_Config.iWindowWidth) / 2;
-			g_Config.iWindowY = (primaryScreenHeight - g_Config.iWindowHeight) / 2;
-			rc.left = g_Config.iWindowX;
-			rc.top = g_Config.iWindowY;
-			rc.right = rc.left + g_Config.iWindowWidth;
-			rc.bottom = rc.top + g_Config.iWindowHeight;
+		// Then center if necessary. One dimension at a time.
```

Comment:
This code is for when starting the first time - though I guess the visibility check can trigger it later too... Is it currently being annoying for you? I'm not sure the visibility check is a good idea anyway.

Should really switch to the newer fully multimonitor capable APIs (with DPI per monitor etc) instead, but then we have to do dynamic loading to avoid breaking XP...